src/utils/src/bytes.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+macro_rules! read_le {
```

Comment:
I would define the macro such that it generates the reading function. Something like:

```
macro_rules! generate_read_le_fn {
    ($fn_name: ident, $data_type: ty, $type_size: expr) => {
        pub fn $fn_name(input: &[u8]) -> $data_type {
                assert!($type_size == std::mem::size_of::<$data_type>());
            let mut array = [0u8; $type_size];
            for (byte, read) in array.iter_mut().zip(input.iter().cloned()) {
                *byte = read as u8;
            }
            <$data_type>::from_le_bytes(array)
        }
    };
}

generate_read_le_fn!(read_le_u16, u16, 2);
```

The same for the `write_le`